### DISCUSSION TITLE
Bitrise Tophat Extension unexpected error

### DISCUSSION THREAD

**Discussion by barbaraakaeze** [Shopify/tophat#57] [Q&A] [answered]
Trying out the new Bitrise extension and running into an `An unexpected error has occurred" error consistently when trying to download the Bitrise artifact. I added the Bitrise app to "Quick Launch" with the required parameters. When I try launching the app, I get that error pop up. I'm not sure of the source of the error; if the Tophat is unable to make the URL request to Bitrise or unable to download the artifact.

Note: I can successfully make API requests to Bitrise using the same token added to Tophat. And I can also successfully download and launch the app using the Bitrise URL. The issue is using the Bitrise extension

**Comment by lfroms** [comment]
Hey @barbaraakaeze, if you are able to successfully select "Bitrise" as the provider in Tophat's Settings, and your personal access token works, then it likely points to one of the following other points of failure:

- Unexpected JSON response format from Bitrise (if this is the case, we should fix it).
- Error while writing to the filesystem and creating temporary files.
- Generic network failure, such as dropped connections.
- The extension unexpectedly crashed while downloading or processing the artifact.

When you click on the entry in Quick Launch, does it spin for a little bit, or does it fail immediately?

Difficult to know exactly where the fault is in this case though.  An unexpected error in Tophat is likely truly "unexpected."  If you're handy with Xcode it might help to start Tophat there with a debugger attached to see any relevant logs.  If nothing stops there, you can also attach the debugger to `TophatBitriseExtension` when Tophat is running in Xcode to see if it catches anything there.

**Reply by barbaraakaeze** [reply, marked as answer]
Quitting the Tophat app and restarting it fixed the unexpected error issue. 
On relaunch, selecting "Bitrise" as the provider in Tophat's Settings and adding the parameters, clicking on the icon in Quick Launch successfully downloads, installs and launches the Bitrise build.

**Reply by lfroms** [reply]
Good to hear!  Must have been a fluke then.  If it happens again please go ahead and open an issue including any details about your setup and any logs.
